ui/components/component-library/text/README.mdx

```diff
@@ -0,0 +1,335 @@
+import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';
+import ActionableMessage from '../../ui/actionable-message';
+import Text from './text';
+
+# Text
+
+> This Text (fka Typography) component has breaking changes in variant options and the line heights associated to each variant.
+
+Good typography improves readability, legibility and hierarchy of information.
+
+<Canvas>
+  <Story id="ui-components-component-library-text-text-stories-js--default-story" />
+</Canvas>
+
+## Props
+
+<ArgsTable of={Text} />
+
+
+### Variant
+
+Use the `variant` prop and the `TEXT` object from `./ui/helpers/constants/design-system.js` to change the font size of the component.
+
+<Canvas>
+  <Story id="ui-components-component-library-text-text-stories-js--variant" />
+</Canvas>
+
+```jsx
+// If importing from ui/components/app/[YOUR_COMPONENT]/ directory
+import Text from '../../ui/component-library/text';
+import { TEXT } from '../../../helpers/constants/design-system';
+
+
+<Text variant="DISPLAY_MD">display-md</Text>
+<Text variant="HEADING_LG">heading-lg</Text>
+<Text variant="HEADING_MD">heading-md</Text>
+<Text variant="HEADING_SM">heading-sm</Text>
+<Text variant="BODY_MD">body-md</Text>
+<Text variant="BODY_SM">body-sm</Text>
+<Text variant="BODY_XS">body-xs</Text>
+```
+
+### Color
+
+Use the `color` prop and the `TEXT_COLORS` object from `./ui/helpers/constants/design-system.js` to change the color of the `Text` component.
```

Comment:
My preference would be just using COLORS, *however* we must acknowledge the fact that with something that is exported like that it will inevitably be used and we will have to either actively look for usages of it outside of storybook/prop validation  OR just be okay with it.

Reply:
OK. Let's leave it up to the engineer's preference. I think we could improve the Color docs in storybook to include these objects and clarify that COLORS is fine to use but there are specific object available if required. I'll create a ticket.

Reply:
Will work on https://github.com/MetaMask/metamask-extension/issues/15594 this week